src/main/java/com/griddynamics/cd/nrp/internal/uploading/impl/ArtifactUpdateApiClientImpl.java

```diff
@@ -119,14 +185,15 @@ public void onComplete(Future<RestResponse> future) throws InterruptedException
 
     /**
      * Creates jersey HTTP client
-     * @param login Username on the remote server
+     *
+     * @param login    Username on the remote server
      * @param password User's password
      * @return HTTP client
      */
     private Client getClient(String login, String password) {
         ClientConfig config = new DefaultClientConfig();
```

Comment:
Looks like this config doesn't have anything configured.. What we need to set is _relatively_ low timeouts for connection creation and for connection duration. Just to illustrate what may happen:
- Nexus Receiver is behind a firewall and someone just configured FW rules in a wrong way. FW silently drops packages from Nexus Sender.
- Nexus Sender has large timeouts and therefore waits for 30 mins on every HTTP request to the Receiver.

Result: all sending threads are blocked for a _very_ long time and there is no way to fix that without Nexus Sender restart.